src/admin/class-options.php

```diff
@@ -0,0 +1,318 @@
+			],
+			'duplicate_post_types_enabled'                => [
+				'tab'      => 'permissions',
+				'callback' => 'generate_post_types_list',
+			],
+			'duplicate_post_show_original_meta_box'       => [
+				'tab'         => 'display',
+				'fieldset'    => 'show-original',
+				'type'        => 'checkbox',
+				'label'       => __( 'In a metabox in the Edit screen [Classic editor]', 'duplicate-post' ),
+				'value'       => 1,
+				'description' => [
+					__( "You'll also be able to delete the reference to the original item with a checkbox", 'duplicate-post' ),
+				],
+			],
+			'duplicate_post_show_original_column'         => [
+				'tab'         => 'display',
+				'fieldset'    => 'show-original',
+				'type'        => 'checkbox',
+				'label'       => __( 'In a column in the Post list', 'duplicate-post' ),
+				'value'       => 1,
+				'description' => [
+					__( "You'll also be able to delete the reference to the original item with a checkbox in Quick Edit", 'duplicate-post' ),
+				],
+			],
+			'duplicate_post_show_original_in_post_states' => [
+				'tab'      => 'display',
+				'fieldset' => 'show-original',
+				'type'     => 'checkbox',
+				'label'    => __( 'After the title in the Post list', 'duplicate-post' ),
+				'value'    => 1,
+			],
+			'duplicate_post_show_notice'                  => [
+				'tab'   => 'display',
+				'type'  => 'checkbox',
+				'label' => __( 'Show update notice', 'duplicate-post' ),
+				'value' => 1,
+			],
+			'duplicate_post_show_link'                    => [
+				'tab'         => 'display',
+				'fieldset'    => 'show-links',
+				'sub_options' => [
+					'new_draft'         => [
+						'type'  => 'checkbox',
+						'label' => __( 'New Draft', 'duplicate-post' ),
+						'value' => 1,
+					],
+					'clone'             => [
+						'type'  => 'checkbox',
+						'label' => __( 'Clone', 'duplicate-post' ),
+						'value' => 1,
+					],
+					'rewrite_republish' => [
+						'type'  => 'checkbox',
+						'label' => __( 'Rewrite & Republish', 'duplicate-post' ),
+						'value' => 1,
+					],
+				],
+			],
+			'duplicate_post_show_link_in' => [
+				'tab'         => 'display',
+				'fieldset'    => 'show-links-in',
+				'sub_options' => [
+					'row'         => [
+						'type'     => 'checkbox',
+						'label'    => __( 'Post list', 'duplicate-post' ),
+						'value'    => 1,
+					],
+					'adminbar' => [
+						'type'        => 'checkbox',
+						'label'       => __( 'Admin bar', 'duplicate-post' ),
+						'value'       => 1,
+						'description' => __( 'now works on Edit screen too - check this option to use with Gutenberg enabled', 'duplicate-post' ),
+					],
+					'submitbox' => [
+						'type'  => 'checkbox',
+						'label' => __( 'Edit screen', 'duplicate-post' ),
+						'value' => 1,
+					],
+					'bulkactions' => [
+						'type'  => 'checkbox',
+						'label' => __( 'Bulk Actions', 'default' ),
+						'value' => 1,
+					],
+				],
+			],
+		];
+	}
+}
```

Comment:
Lots of duplicate code here, i would suggest using a builder pattern with defaults set to:
```
'tab'      => 'what-to-copy',
'fieldset' => 'elements-to-copy',
'type'     => 'checkbox',
'label'    => __( 'Title', 'default' ),
'value'    => 1,
```

Reply:
Sure, there's _some_ duplication here, but seeing as this is already a massive change / improvement from what it previously was, I feel like this should be addressed at a later time. Additionally, this approach was pre-approved by @herregroen for the time being.

Reply:
Discussed with @herregroen , and we concluded that this is not worth implementing 👍🏻